provider/equinix/environ.go

```diff
@@ -187,3 +199,245 @@ func (e *environ) Region() (simplestreams.CloudSpec, error) {
 		Endpoint: e.cloud.Endpoint,
 	}, nil
 }
+
+// if values tag and state are left empty it will return all instances
+func (e *environ) getPacketInstancesByTag(tags map[string]string) ([]instances.Instance, error) {
+	toReturn := []instances.Instance{}
+	deviceTags := set.NewStrings()
+
+	for k, v := range tags {
+		deviceTags.Add(fmt.Sprintf("%s=%s", k, v))
+	}
+	devices, _, err := e.equinixClient.Devices.List(e.cloud.Credential.Attributes()["project-id"], nil)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	for _, d := range devices {
+		cp := d
+		cpTags := set.NewStrings(cp.Tags...)
+		if !deviceTags.Intersection(cpTags).IsEmpty() {
+			toReturn = append(toReturn, &equinixDevice{e, &cp})
+		}
+	}
+
+	return toReturn, nil
+}
+
+func (e *environ) getSubnetsToZoneMap(ctx context.ProviderCallContext, args environs.StartInstanceParams) ([]string, error) {
+	var subnetIDs []string
+	for _, subnetList := range args.SubnetsToZones {
+		for subnetID := range subnetList {
+			packetSubnetID := strings.TrimPrefix(subnetID.String(), "subnet-")
+			subnetIDs = append(subnetIDs, packetSubnetID)
+		}
+	}
+
+	return subnetIDs, nil
+}
+
+// supportedInstanceTypes returns the instance types supported by Equnix Metal.
+func (e *environ) supportedInstanceTypes() ([]instances.InstanceType, error) {
+	opt := &packngo.ListOptions{
+		Includes: []string{"available_in"},
+	}
+	plans, _, err := e.equinixClient.Plans.List(opt)
+	if err != nil {
+		return nil, errors.Annotate(err, "retrieving supported instance types")
+	}
+
+	var instTypes []instances.InstanceType
+nextPlan:
+	for _, plan := range plans {
+		if !validPlan(plan, e.cloud.Region) {
+			logger.Debugf("Plan %s not valid in facility %s", plan.Name, e.cloud.Region)
+			continue
+		}
+
+		var instArch string
+		switch {
+		case strings.HasSuffix(plan.Name, ".x86"):
+			instArch = arch.AMD64
+		case strings.HasSuffix(plan.Name, ".arm"):
+			instArch = arch.ARM64
```

Comment:
Does equinix support any other arches that we need to include here?

Reply:
Also, I see a `getArchitectureFromPlan` helper below that you can use to replace the switch here.

Reply:
I can't use it because the logic for `getArchitectureFromPlan` is a bit different, it returns a default architecture instead of continuing. I can change the behavior for that...